cmd/webhook.go

```diff
@@ -169,14 +168,86 @@ func updateAnnotation(target map[string]string, added map[string]string) (patch
 	return patch
 }
 
+// This will ADD a volumeMount to the user container spec
+func updateWorkingVolumeMounts(targetContainerSpec []corev1.Container, bucketName string, isFirst bool, namespace string) (patch []patchOperation) {
+	for key := range targetContainerSpec {
+		// if there is an envVar that has NB_PREFIX in it then we are in the right one
+		for envVars := range targetContainerSpec[key].Env {
+			if targetContainerSpec[key].Env[envVars].Name == "NB_PREFIX" {
+				var mapSlice []M
+				valueA := M{"name": "fuse-csi-ephemeral-" + bucketName + "-" + namespace, "mountPath": "/home/jovyan/" + bucketName,
+					"readOnly": false, "mountPropagation": "HostToContainer"}
+				mapSlice = append(mapSlice, valueA)
+				if isFirst {
+					patch = append(patch, patchOperation{
+						Op: "add",
+						// the path for only the first value
+						Path:  "/spec/containers/0/volumeMounts",
+						Value: mapSlice,
+					})
+				} else {
+					patch = append(patch, patchOperation{
+						Op: "add",
+						// Now that there is one that has created an array, this can just go after it.
+						Path:  "/spec/containers/0/volumeMounts/-",
+						Value: valueA,
+					})
+				}
+			}
+		}
+	}
+	return patch
+}
+
 // create mutation patch for resoures
-func createPatch(pod *corev1.Pod, sidecarConfig *Config, annotations map[string]string) ([]byte, error) {
+func createPatch(pod *corev1.Pod, oldSidecarConfig *Config, annotations map[string]string) ([]byte, error) {
```

Comment:
Maybe a better name for oldSidecarConfig would be something like "sidecarConfigTemplate"?